### DISCUSSION TITLE
ScriptReference/RaycastHit2D

### DISCUSSION THREAD

**Discussion by giscus** [Developer-Notes-Extension/Unity-Notes#536] [Notes]
# ScriptReference/RaycastHit2D

https://docs.unity3d.com/ScriptReference/RaycastHit2D.html

**Comment by boxfriend** [comment]
Standard practice for checking whether a raycast was successful is to check if the `hit.collider` is not null. However RaycastHit2D has an implicit cast to bool that does this for you

<https://github.com/Unity-Technologies/UnityCsReference/blob/master/Modules/Physics2D/ScriptBindings/Physics2D.bindings.cs#L3204>
